### PR TITLE
SALTO-2870: Remove consumerKey from ConnectedAppOauthConfig

### PR DESCRIPTION
**Pull request by tamtamirr**
Removing the `consumerKey` from ConnectedApp instances.

---

- Noise Suppression: https://github.com/salto-io/salto_private/pull/5193
- Workspace Diff: https://github.com/salto-io/tamir-sf/compare/b83f8d10236a195ffb9fe5c5fb2103285c67e3ba..38cf965fefd22f8d1aae25b91157e0518946ee8d

---
_Release Notes_: 
Salesforce Adapter:
- Support deployment of `ConnectedApp` with `OAuth Config` enabled.

---
_User Notifications_: 
Salesforce:
- Removal of `consumerKey` from existing `ConnectedApp` instances.

### CODE REVIEW

File: packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { InstanceElement } from '@salto-io/adapter-api'
+import filterCreator from '../../src/filters/remove_connected_app_oauth_config_consumer_key'
+import { mockTypes } from '../mock_elements'
+
+const filter = filterCreator()
+
+describe(filter.name, () => {
```
Comment (Uri-Bar) [inline packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts:22]: did you ment to put the filter name here?

File: packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      connectedAppInstance = new InstanceElement(
+        'testApp',
+        mockTypes.ConnectedApp,
+        {
+          oauthConfig: {
+            consumerKey: 'testKey123435',
+          },
+        }
+      )
+      expect(connectedAppInstance.value.oauthConfig.consumerKey).toBeDefined()
```
Comment (Uri-Bar) [inline packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts:36]: do we use except inside of before each?

Comment (tamtamirr) [reply packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts:36]: Yes, to validate the test itself actually handles a "valid scenario". A failure in `beforeEach` means the test itself is incorrect

File: packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/test/filters/remove_connected_app_oauth_config_consumer_key.test.ts:22]: THis is the filter name :)

**Review by Uri-Bar — APPROVED** [review]
thanks for that

**Comment by tamtamirr** [comment]
After discussing with @ori-moisis, closing this one as it's not a a common use-case and removing the `consumerKey` misses the intention.

**Pull request closed without merging**